sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client_helpers.py

```diff
@@ -950,9 +950,10 @@ def _upload_page_options(
     length: int,
     **kwargs: Any
 ) -> Dict[str, Any]:
+    page = page[:length]
     if isinstance(page, str):
         page = page.encode(kwargs.pop('encoding', 'UTF-8'))
-    page = page[:length]
```

Comment:
As an aside- we should probably document this in a changelog? Alongside the type "constriction" that occurred 😄

Reply:
Yes, I have a few changelog entries ready to go for whenever this actually ships.